packages/open-next/src/build/patch/codePatcher.ts

```diff
@@ -0,0 +1,114 @@
+import * as buildHelper from "../helper.js";
+import * as fs from "node:fs/promises";
+
+// Either before or after should be provided, otherwise just use the field directly
+interface VersionedField<T> {
+  before?: `${number}.${number}.${number}`;
+  after?: `${number}.${number}.${number}`;
+  field: T;
+}
+
+//TODO: create a version of this that would use ast-grep
+type PatchCodeFn = (args: {
+  // The content of the file that needs to be patched
+  code: string;
+  // The final path of the file that needs to be patched
+  filePath: string;
+  // All js files that will be included in this specific server function
+  tracedFiles: string[];
+  // All next.js manifest that are present at runtime - Key relative to `.next` folder
+  manifests: Record<string, any>;
+}) => Promise<string>;
+
+export interface CodePatcher {
+  name: string;
+  filter: RegExp | VersionedField<RegExp>[];
```

Comment:
We use `filter` to match ESBuild but `pathFilter` would be nice IMO (vs `contentFilter`)